src/App.tsx

```diff
@@ -1,25 +1,22 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
+
+import Header from "./Header";
+import ProductsPage from "./pages/ProductsPage";
+import SomePage2 from "./pages/SomePage2";
+import "./App.css";
 
 function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <BrowserRouter>
+        <Header />
+        <Routes>
+          <Route path="/SomePage1" element={<ProductsPage />} />
+          <Route path="/SomePage2/:id" element={<SomePage2 />} />
+          <Route path="*" element={<Navigate to="/SomePage1" replace />} />
```

Comment:
пути необходимо делать человекопонятными, чтобы при шеринге ссылке, понятно куда переходишь
```suggestion
          <Route path="/products" element={<ProductsPage />} />
          <Route path="/product/:id" element={<SomePage2 />} />
          <Route path="*" element={<Navigate to="/products" replace />} />
```

С названиями компонентов то же самое, легче ориентироваться в коде

Reply:
переделал